pkg/actuators/machine/utils.go

```diff
@@ -88,7 +81,8 @@ func getInstances(machine *machinev1.Machine, client awsclient.Client, instanceS
 
 	// Query instances with our machine's name, and in running/pending state.
 	request := &ec2.DescribeInstancesInput{
-		Filters: requestFilters,
+		Filters:     requestFilters,
+		InstanceIds: []*string{machine.Spec.ProviderID},
```

Comment:
how would this work? We'd need something to "decode" the spec.ProviderID to get only the "i-1234" part of it?